src/PhotosphereEditor.tsx

```diff
@@ -49,13 +58,104 @@ function PhotosphereEditor({
 
   const [showAddFeatures, setShowAddFeatures] = useState(false);
   const [showChangeFeatures, setShowChangeFeatures] = useState(false);
+  const [showRemoveFeatures, setShowRemoveFeatures] = useState(false);
 
   const [newName, setNewName] = useState(""); // State to hold the new name
   const [newBackground, setNewBackground] = useState("");
   const [audio, setAudio] = useState<string>("");
 
+  const [showRemovePhotosphere, setShowRemovePhotosphere] = useState(false);
+  const [showRemoveNavMap, setShowRemoveNavMap] = useState(false);
+  const [showRemoveHotspot, setShowRemoveHotspot] = useState(false);
+  const [hotspotToRemove, setHotspotToRemove] = useState<string | null>(null);
   console.log(vfe);
 
+  // Change URL to reflect current photosphere
+
+  function handleRemoveHotspotClick(hotspotToRemove: string) {
+    setHotspotToRemove(hotspotToRemove);
+    setShowRemoveHotspot(true);
+  }
+
+  function handleRemoveHotspotConfirm() {
+    if (hotspotToRemove) {
+      const updatedPhotosphere = { ...vfe.photospheres[photosphereID] };
+
+      // Create a new object without the hotspot to remove
+      const remainingHotspots = Object.fromEntries(
+        Object.entries(updatedPhotosphere.hotspots).filter(
+          ([key]) => key !== hotspotToRemove,
+        ),
+      );
+
+      // Update the photosphere with the remaining hotspots
+      const updatedPhotosphereWithHotspots = {
+        ...updatedPhotosphere,
+        hotspots: remainingHotspots,
+      };
+
+      const updatedVFE = {
+        ...vfe,
+        photospheres: {
+          ...vfe.photospheres,
+          [photosphereID]: updatedPhotosphereWithHotspots,
+        },
+      };
+
+      //setVFE(updatedVFE); // Update the local VFE state
+      onUpdateVFE(updatedVFE); // Propagate the change to the parent component
+      setShowRemoveHotspot(false); // Close the RemoveHotspot component
+      setHotspotToRemove(null);
+      setUpdateTrigger((prev) => prev + 1);
+    }
+  }
+
+  function handleRemovePhotosphere(photosphereId: string) {
+    if (!photosphereId) {
+      alert("Photosphere not found.");
+      return;
+    }
+
+    // Create a new object without the removed photosphere
+    const remainingPhotospheres = Object.fromEntries(
+      Object.entries(vfe.photospheres).filter(([key]) => key !== photosphereId),
+    );
+
+    const nextPhotosphereId = Object.keys(remainingPhotospheres)[0];
+
+    const newDefaultPhotosphereID =
+      photosphereId === vfe.defaultPhotosphereID
+        ? nextPhotosphereId
+        : vfe.defaultPhotosphereID;
+
+    if (Object.keys(remainingPhotospheres).length === 0) {
+      // No more photospheres available
+      alert("No more photospheres available.");
+      navigate("/"); // Redirect to home?
+      return;
+    }
```

Comment:
I think it would be a better user experience if the button just tells the user that the last photosphere can't be deleted without redirecting to home.

Reply:
Maybe we could have a new window pop up to ask like: "you will return to home page if you delete this photosphere. Do you wish to continue?" and a yes-no buttons? But in that case, then if the user hit no, then we not gonna delete the photosphere.

Or we can just display a message: "you will return to home page after delete this photosphere" and return back to the home page without giving them the options to choose.

Reply:
It looks like the current behavior redirects to home page but doesn't actually delete the final photosphere. After going back to the home page and clicking on the VFE that the user was just redirected from, the final photosphere is still there.

Maybe the entire VFE should be deleted when there are no more photospheres?

Reply:
Yes, you are correct. The entire VFE should be deleted if there is no more photosphere.